geopandas/geoseries.py

```diff
@@ -534,7 +534,31 @@ def _constructor(self):
     def _constructor_expanddim(self):
         from geopandas import GeoDataFrame
 
-        return GeoDataFrame
+        def expanddim(data=None, index=None, crs=None, **kwargs):
+            # if geometry is somehow supplied, ignore it because we don't need it
+            kwargs.pop("geometry", None)
+            if isinstance(data, GeoSeries):
+                #  if GeoSeries has no name, we make the geometry name geometry
+                #  (instead of 0)
+                if data.name is None:
+                    data = {"geometry": data}
+                    geo_col_name = "geometry"
+                else:
+                    geo_col_name = data.name
+            else:  # Dict of {name : (Geo)Series pairs}
+                # if only one key, this is not ambiguous, if more than one
+                # i.e. from pd.concat([GeoSeries, GeoSeries]) the geom col comes
+                # from the left (same as for multiple gdfs in concat)
+                geo_col_name = list(data.keys())[0]
+
+            df = GeoDataFrame(
+                data=data, index=index, crs=crs, geometry=geo_col_name, **kwargs
+            )
+            return df
+
+        expanddim._get_axis_number = GeoDataFrame._get_axis_number
```

Comment:
Why is this exactly needed? (for some compatibility with older pandas? if so, can you add a comment about it?)

Reply:
I'm actually not sure why now, whether I needed it in an initial version or if I pulled it mistakenly from one of the branches on my fork with too many changes at once. In any case it doesn't seem necessary for the current iteration.

Reply:
Ah, I'm not crazy, I was checking if it was needed for old pandas, but it's actually needed for new pandas, concat needs this: 
https://github.com/pandas-dev/pandas/blob/e39ea3024cebb4e7a7fd35972a44637de6c41650/pandas/core/reshape/concat.py#L465-L469

Which seems to be introduced from pandas 1.1
https://github.com/pandas-dev/pandas/commit/f9e4c8c84bcef987973f2624cc2932394c171c8c.

I'll make a note in the code.